lib/matplotlib/colorbar.py

```diff
@@ -621,37 +610,28 @@ def _add_solids_patches(self, X, Y, C, mappable):
 
     def _do_extends(self, extendlen):
         """
-        Make adjustments of the inner axes for the extend triangles (or
-        rectanges) and add them as patches.
+        Add the extend tri/rectangles on the outside of the axes.
         """
         # extend lengths are fraction of the *inner* part of colorbar,
         # not the total colorbar:
-        elower = extendlen[0] if self._extend_lower() else 0
-        eupper = extendlen[1] if self._extend_upper() else 0
-        total_len = eupper + elower + 1
-        elower = elower / total_len
-        eupper = eupper / total_len
-        inner_length = 1 / total_len
-
-        # make the inner axes smaller to make room for the extend rectangle
-        top = elower + inner_length
+        bot = extendlen[0] if self._extend_lower() else 0
+        bot = -bot
+        top = extendlen[1] if self._extend_upper() else 0
+        top = top + 1
```

Comment:
These could be compressed a bit:
```suggestion
        bot = 0 - (extendlen[0] if self._extend_lower() else 0)
        top = 1 + (extendlen[1] if self._extend_upper() else 0)
```
I realize that's still a bit redundant, but I kept in the left-most terms because it's clearer to me what's going on (i.e., that we're explicitly going outside the normal Axes transform of 0-1). Maybe that's too much though.